presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -315,8 +315,8 @@ public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Opt
     public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Optional<Integer> hashChannel, Optional<JoinFilterFunction> joinFilterFunction)
     {
         try {
-            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels)
-                    .createPagesHashStrategy(ImmutableList.copyOf(channels), hashChannel);
+            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels, joinFilterFunction.map(f -> f.getClass()))
```

Comment:
`JoinFilterFunction::getClass`

IntelliJ should warn you on this line: "can be replaced with method reference". If it doesn't, please turn the warning on.